### ISSUE TITLE
--py3k should ignore code gated to Python 2

### ISSUE THREAD

**Issue by rowillia** [PyCQA/pylint#1194]
It's quite common for a codebase to maintain 2and3 compatibility through gates in the code.  Pylint should recognize these gates and ignore code explicitly gated to Python 2.

For example:

```python
if six.PY2:
  # Ignore this code

if six.PY3:
  # Some totally sweet Python 3 code
else:
  # Ignore this code

if sys.version_info[0] == 2:
  # Ignore this code

if sys.version_info >= (3, 0):
  # Python 3 code
else:
  # Ignore this code
```

The easiest way to do this would blow up execution times by walking the tree parent by parent looking for a gate.  Unfortunately, I don't know enough about astroid to figure out how to efficiently accomplish this.  The main pointer I'd love is how I would either:

1) Tell the visitor to skip all of the children of a given node.
2) Get a callback when exiting a given node.

**Comment by rogalski** [comment]
@rowillia  @PCManticore does c002c48a9b8d98b43262e045243af2a80cb8a1f3 means that this issue can be closed?

**Comment by PCManticore** [comment]
I think so, yes. @rowillia let me know if this is good enough for you.
